### PR TITLE
refactor(storybook): consolidate storybook and component types

### PR DESCRIPTION
**Pull request by aPreciado88**
**Related Issue:** [#9418](https://github.com/Esri/calcite-design-system/issues/9418)

## Summary

- Consolidates all types contained in `packages/calcite-components/.storybook/interfaces.ts` and `packages/calcite-components/src/components/interfaces.ts`
- Updates components to only pick up types that they support.

### CODE REVIEW

File: packages/calcite-components/src/components/tile-group/tile-group.tsx [hunk 1]
```diff
@@ -53,7 +53,10 @@ export class TileGroup implements InteractiveComponent, SelectableGroupComponent
    *
    * Use `"horizontal"` for rows, and `"vertical"` for a single column.
    */
-  @Prop({ reflect: true }) layout: Exclude<Layout, "grid"> = "horizontal";
+  @Prop({ reflect: true }) layout: Exclude<
```
Comment (jcfranco) [inline packages/calcite-components/src/components/tile-group/tile-group.tsx:56]: Based on this PR, I think it's better if we change this to `Extract` instead. Applies to other relevant exclusions.

File: packages/calcite-components/src/components/block-section/block-section.tsx [hunk 2]
```diff
@@ -14,15 +14,14 @@ import {
 import { focusFirstTabbable, toAriaBoolean } from "../../utils/dom";
 import { isActivationKey } from "../../utils/key";
 import { connectLocalized, disconnectLocalized, LocalizedComponent } from "../../utils/locale";
-import { FlipContext } from "../interfaces";
```
Comment (jcfranco) [inline packages/calcite-components/src/components/block-section/block-section.tsx:17]: Can we submit these import-only changes separately?

File: packages/calcite-components/src/components/carousel/interfaces.ts [hunk 3]
```diff
@@ -1,4 +1,2 @@
-export type ArrowType = "inline" | "edge" | "none";
```
Comment (jcfranco) [inline packages/calcite-components/src/components/carousel/interfaces.ts:1]: We can leave component-specific types where they are. Anything that's shared should go in `src/components/interfaces.ts`. Some new types might need to be moved to component-level `interfaces.ts` files.

**Review by jcfranco — APPROVED** [review]
Aside from a few cleanup comments, this LGTM!

@anveshmekala @driskull @Elijbet @eriklharper @macandcheese can you give this a quick look? 👀

**Comment by aPreciado88** [comment]
@jcfranco I updated the PR based on your comments. I'll create a new PR to submit the import-only [changes](https://github.com/Esri/calcite-design-system/pull/9500/files/9ad2ad980620cf1c89f54682f2a30c04397f176b#diff-a6a86252475c12915cd4c402b255ec9c53f5d4d0f0e9ed4be33ab2c3f4bafe2c)

**Review by alisonailea — APPROVED** [review]

**Pull request merged**